src/main/kotlin/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/MtSharedTableBackupManager.kt

```diff
@@ -0,0 +1,456 @@
+
+    override fun deleteBackup(id: String): MtBackupMetadata? {
+        var continuationToken: String? = null
+        var deleteCount = 0
+        do {
+            val listBucketResult: ListObjectsV2Result = s3.listObjectsV2(
+                    ListObjectsV2Request()
+                            .withBucketName(s3BucketName)
+                            .withContinuationToken(continuationToken)
+                            .withPrefix("$backupDir/$id/"))
+            continuationToken = listBucketResult.continuationToken
+            if (listBucketResult.objectSummaries.size > 0) {
+                deleteCount += listBucketResult.objectSummaries.size
+                s3.deleteObjects(DeleteObjectsRequest(s3BucketName).withKeys(
+                        listBucketResult.objectSummaries.stream()
+                                .map { DeleteObjectsRequest.KeyVersion(it.key) }
+                                .collect(Collectors.toList()) as List<DeleteObjectsRequest.KeyVersion>).withQuiet(true))
+            }
+        } while (listBucketResult.isTruncated)
+        if (deleteCount > 0) {
+            logger.info("$id: Deleted $deleteCount backup files for $id")
+        }
+        val ret = getBackup(id)
+        s3.deleteObject(DeleteObjectRequest(s3BucketName, getBackupMetadataFile(id)))
+        return ret
+    }
+
+    override fun getTenantTableBackupFromArn(backupArn: String): TenantTableBackupMetadata {
+        if (!isTenantTableArn(backupArn)) {
+            throw IllegalArgumentException("$backupArn does not include tenant-table specifier.")
+        }
+        val tenantTableParts = backupArn.split(':')
+        return TenantTableBackupMetadata(tenantTableParts[0], tenantTableParts[1], tenantTableParts[2])
+    }
+
+    private fun isTenantTableArn(backupArn: String): Boolean = backupArn.split(':').size == 3
+
+    override fun getBackupArnForTenantTableBackup(tenantTable: TenantTableBackupMetadata): String =
+            "${tenantTable.backupName}:${tenantTable.tenantId}:${tenantTable.virtualTableName}"
+
+    private fun commitTenantTableMetadata(
+        backupId: String,
+        tenantTableMetadatas: MtTableDescriptionRepo.ListMetadataResult
+    ) {
+        for (tenantTableMetadata in tenantTableMetadatas.createTableRequests) {
+            val tenantTableMetadataJson = gson.toJson(tenantTableMetadata.createTableRequest).toByteArray(charset)
+            val objectMetadata = ObjectMetadata()
+            objectMetadata.contentLength = tenantTableMetadataJson.size.toLong()
+            objectMetadata.contentType = "application/json"
+            val putObjectReq = PutObjectRequest(s3BucketName,
+                    getTenantTableMetadataFile(backupId, TenantTable(tenantTableMetadata.createTableRequest.tableName, tenantTableMetadata.tenantName)),
+                    gson.toJson(tenantTableMetadata.createTableRequest).byteInputStream(charset),
+                    objectMetadata)
+            s3.putObject(putObjectReq)
```

Comment:
Note: reads may not return the latest metadata after updating it: https://codeburst.io/quick-explanation-of-the-s3-consistency-model-6c9f325e3f82